### PR TITLE
Openssl: Allow to disable TLSv1.3 and force using TLSv1.3 in configuration options

### PR DESCRIPTION
**Pull request by mitzkia**
This patch set allows to use TLS v1.3 protocol version in TLS contexts and in ssl_version() option 

- ssl_version() option has been expanded with "tlsv1_3" value
- with this value syslog-ng can force using TLS v1.3 protocol with server
```
http(
    url("https://localhost:8081") 
    ssl_version("tlsv1_3")
)
```

- ssl_option() option has been expanded with "no-tlsv13" value
- with this value syslog-ng can disable TLS v1.3 protocol usage
```
    network(
         "172.16.177.147" port(6514)
        transport("tls")
        tls(
            ca-dir("/etc/syslog-ng/ca.d")
            key-file("/etc/syslog-ng/cert.d/clientkey.pem")
            cert-file("/etc/syslog-ng/cert.d/clientcert.pem")
            ssl-options(no-tlsv13)
        )
    );
```

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/tlscontext.c [hunk 1]
```diff
@@ -931,6 +939,14 @@ tls_context_set_ssl_options_by_name(TLSContext *self, GList *options)
         self->ssl_options |= TSO_NOTLSv11;
       else if (strcasecmp(l->data, "no-tlsv12") == 0 || strcasecmp(l->data, "no_tlsv12") == 0)
         self->ssl_options |= TSO_NOTLSv12;
+#endif
+#ifdef SSL_OP_NO_TLSv1_3
+      else if (strcasecmp(l->data, "no-tlsv11") == 0 || strcasecmp(l->data, "no_tlsv11") == 0)
+        self->ssl_options |= TSO_NOTLSv11;
+      else if (strcasecmp(l->data, "no-tlsv12") == 0 || strcasecmp(l->data, "no_tlsv12") == 0)
+        self->ssl_options |= TSO_NOTLSv12;
```
Comment (MrAnno) [inline lib/tlscontext.c]: Only the TLS1.3 check should be added.

**Comment by MrAnno** [comment]
Please rephrase the title and the commit messages.
TLS 1.3 works without this PR.

You've added the `no-tlsv13` option to make it possible to **disable** TLS1.3, and the `http()` option `tlsv1_3` is for forcing TLS 1.3-only.

File: lib/tlscontext.c [hunk 2]
```diff
@@ -457,6 +457,14 @@ tls_context_setup_ssl_options(TLSContext *self)
       if(self->ssl_options & TSO_NOTLSv12)
         ssl_options |= SSL_OP_NO_TLSv1_2;
 #endif
+#ifdef SSL_OP_NO_TLSv1_3
+      if(self->ssl_options & TSO_NOTLSv11)
+        ssl_options |= SSL_OP_NO_TLSv1_1;
+      if(self->ssl_options & TSO_NOTLSv12)
+        ssl_options |= SSL_OP_NO_TLSv1_2;
```
[the remark this answers is not in the document]
Comment (mitzkia) [reply lib/tlscontext.c]: Thanks for the review note. I have fixed it.

File: lib/tlscontext.c (hunk 1 above)
Comment (mitzkia) [reply lib/tlscontext.c]: Thanks for the review note. I have fixed it.

**Comment by mitzkia** [comment]
@MrAnno Thanks for the review notes, I have fixed them.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**